core/services/relay/evm/chain_writer.go

```diff
@@ -30,11 +33,15 @@ type ChainWriterService interface {
 // Compile-time assertion that chainWriter implements the ChainWriterService interface.
 var _ ChainWriterService = (*chainWriter)(nil)
 
-func NewChainWriterService(logger logger.Logger, client evmclient.Client, txm evmtxmgr.TxManager, config types.ChainWriterConfig) (ChainWriterService, error) {
+func NewChainWriterService(logger logger.Logger, client evmclient.Client, txm evmtxmgr.TxManager, estimator gas.EvmFeeEstimator, config types.ChainWriterConfig) (ChainWriterService, error) {
 	w := chainWriter{
 		logger: logger,
 		client: client,
 		txm:    txm,
+		ge: CWGasEstimator{
+			estimator:   estimator,
+			maxGasPrice: config.MaxGasPrice,
```

Comment:
do we have what maxGasPrice is and how it's used documented?

Reply:
`maxGasPrice` is used by the Evm Estimator as an override of the `feeCap` for the transaction. I added a comment in the struct declaration explaining this